File: UsuariosAPI/src/main/java/com/clinica/api/personal_service/controller/PersonalController.java
package com.clinica.api.personal_service.controller;

import com.clinica.api.personal_service.model.Doctor;
import com.clinica.api.personal_service.service.PersonalService;
import jakarta.persistence.EntityNotFoundException;
import java.util.List;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/personal")
public class PersonalController {

    private final PersonalService personalService;

    public PersonalController(PersonalService personalService) {
        this.personalService = personalService;
    }

    @GetMapping
    public ResponseEntity<List<Doctor>> getAllDoctores() {
        List<Doctor> doctores = personalService.findAllDoctores();
        if (doctores.isEmpty()) {
            return ResponseEntity.noContent().build();
        }
        return ResponseEntity.ok(doctores);
    }

    @GetMapping("/{id}")
    public ResponseEntity<Doctor> getDoctorById(@PathVariable Long id) {
        try {
            return ResponseEntity.ok(personalService.findDoctorById(id));
        } catch (EntityNotFoundException ex) {
            return ResponseEntity.notFound().build();
        }
    }

    @PostMapping
    public ResponseEntity<Doctor> createDoctor(@RequestBody Doctor doctor) {
        Doctor nuevoDoctor = personalService.saveDoctor(doctor);
        return ResponseEntity.status(HttpStatus.CREATED).body(nuevoDoctor);
    }

    @PutMapping("/{id}")
    public ResponseEntity<Doctor> updateDoctor(@PathVariable Long id, @RequestBody Doctor doctorDetails) {
        try {
            Doctor existente = personalService.findDoctorById(id);
            existente.setSueldo(doctorDetails.getSueldo());
            existente.setUsuario(doctorDetails.getUsuario());
            Doctor actualizado = personalService.saveDoctor(existente);
            return ResponseEntity.ok(actualizado);
        } catch (EntityNotFoundException ex) {
            return ResponseEntity.notFound().build();
        }
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteDoctor(@PathVariable Long id) {
        try {
            personalService.deleteDoctorById(id);
            return ResponseEntity.noContent().build();
        } catch (EntityNotFoundException ex) {
            return ResponseEntity.notFound().build();
        }
    }
}

File: personal-service/src/main/kotlin/com/clinica/api/personal_service/controller/PersonalController.kt
package com.clinica.api.personal_service.controller

import com.clinica.api.personal_service.model.Doctor
import com.clinica.api.personal_service.service.PersonalService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/personal")
class PersonalController(private val personalService: PersonalService) {

    @GetMapping
    fun getAllDoctores(): ResponseEntity<List<Doctor>> {
        val doctores = personalService.findAllDoctores()
        return if (doctores.isNotEmpty()) {
            ResponseEntity.ok(doctores)
        } else {
            ResponseEntity.noContent().build()
        }
    }

    @GetMapping("/{id}")
    fun getDoctorById(@PathVariable id: Long): ResponseEntity<Doctor> {
        return try {
            val doctor = personalService.findDoctorById(id)
            ResponseEntity.ok(doctor)
        } catch (e: Exception) {
            ResponseEntity.notFound().build()
        }
    }

    @PostMapping
    fun createDoctor(@RequestBody doctor: Doctor): ResponseEntity<Doctor> {
        val nuevoDoctor = personalService.saveDoctor(doctor)
        return ResponseEntity(nuevoDoctor, HttpStatus.CREATED)
    }

    @PutMapping("/{id}")
    fun updateDoctor(@PathVariable id: Long, @RequestBody doctorDetails: Doctor): ResponseEntity<Doctor> {
        return try {
            val doctorExistente = personalService.findDoctorById(id)
            // Actualizamos los campos que se pueden modificar
            doctorExistente.sueldo = doctorDetails.sueldo
            doctorExistente.usuario = doctorDetails.usuario // Se asume que el objeto usuario viene completo
            val doctorActualizado = personalService.saveDoctor(doctorExistente)
            ResponseEntity.ok(doctorActualizado)
        } catch (e: Exception) {
            ResponseEntity.notFound().build()
        }
    }

    @DeleteMapping("/{id}")
    fun deleteDoctor(@PathVariable id: Long): ResponseEntity<Void> {
        return try {
            personalService.deleteDoctorById(id)
            ResponseEntity.noContent().build()
        } catch (e: Exception) {
            ResponseEntity.notFound().build()
        }
    }
}